terraform/azure/modules/hana_node/main.tf

```diff
@@ -364,62 +364,60 @@ locals {
 }
 
-resource "azurerm_virtual_machine" "hana" {
-  count                            = var.hana_count
-  name                             = "${var.name}${format("%02d", count.index + 1)}"
-  location                         = var.az_region
-  resource_group_name              = var.resource_group_name
-  network_interface_ids            = [element(azurerm_network_interface.hana.*.id, count.index)]
-  availability_set_id              = var.common_variables["hana"]["ha_enabled"] ? azurerm_availability_set.hana-availability-set[0].id : null
-  vm_size                          = var.vm_size
-  delete_os_disk_on_termination    = true
-  delete_data_disks_on_termination = true
-
-  storage_os_disk {
-    name              = "disk-${var.name}${format("%02d", count.index + 1)}-Os"
-    caching           = "ReadWrite"
-    create_option     = "FromImage"
-    managed_disk_type = "Premium_LRS"
-  }
+resource "azurerm_managed_disk" "hana_data_disk" {
+  count                = var.hana_count * local.disks_number
+  name                 = "disk-${var.name}${format("%02d", floor(count.index / local.disks_number) + 1)}-Data${format("%02d", count.index % local.disks_number + 1)}"
+  location             = var.az_region
+  resource_group_name  = var.resource_group_name
+  storage_account_type = element(local.disks_type, count.index % local.disks_number)
+  create_option        = "Empty"
+  disk_size_gb         = element(local.disks_size, count.index % local.disks_number)
+}
 
-  storage_image_reference {
-    id        = var.sles4sap_uri != "" ? join(",", azurerm_image.sles4sap.*.id) : ""
-    publisher = var.sles4sap_uri != "" ? "" : module.os_image_reference.publisher
-    offer     = var.sles4sap_uri != "" ? "" : module.os_image_reference.offer
-    sku       = var.sles4sap_uri != "" ? "" : module.os_image_reference.sku
-    version   = var.sles4sap_uri != "" ? "" : module.os_image_reference.version
+resource "azurerm_virtual_machine_data_disk_attachment" "hana_data_disk_attachment" {
+  count              = var.hana_count * local.disks_number
+  managed_disk_id    = azurerm_managed_disk.hana_data_disk[count.index].id
+  virtual_machine_id = azurerm_linux_virtual_machine.hana[floor(count.index / local.disks_number)].id
+  lun                = count.index % local.disks_number
+  caching            = element(local.disks_caching, count.index % local.disks_number)
+}
+
+resource "azurerm_linux_virtual_machine" "hana" {
+  count                 = var.hana_count
+  name                  = "${var.name}${format("%02d", count.index + 1)}"
+  location              = var.az_region
+  resource_group_name   = var.resource_group_name
+  network_interface_ids = [element(azurerm_network_interface.hana.*.id, count.index)]
+  availability_set_id   = var.common_variables["hana"]["ha_enabled"] ? azurerm_availability_set.hana-availability-set[0].id : null
+  size                  = var.vm_size
+
+  admin_username = var.common_variables["authorized_user"]
+  admin_ssh_key {
```

Comment:
is no more 
```
path = "/home/${var.common_variables["authorized_user"]}/.ssh/authorized_keys"
```

needed or supported?

Reply:
It's not needed OR supported. From the documentation:

NOTE:
The Azure VM Agent only allows creating SSH Keys at the path /home/{username}/.ssh/authorized_keys - as such this public key will be written to the authorized keys file.